### DISCUSSION TITLE
如何实现后端推送数据？

### DISCUSSION THREAD

**Discussion by hanbingzi** [opensumi/core#3339] [Q&A] [answered]
我参考官网，使用了如下方式：

```ts
@Injectable()
export class TodoNodeService extends RPCService implements ITodoNodeService {
  showMessage = (message: string) => {
    // 这里的 this.rpcClient![0] 可以直接获取到通信通道下的 proxy 实例
    this.rpcClient![0].onMessage(`I got you message, echo again. ${message}`);
  };
}
```
这样往browser端推送数据，但browser端接收不到，不知道这样写正确吗

**Comment by bytemain** [comment, marked as answer]
通信的例子是运行良好的，可以参考：https://github.com/opensumi/opensumi-module-samples/blob/main/modules/connection/browser/index.ts#L21

请检查一下 browser 层的 Module 中是否声明了相应的 clientToken:
![CleanShot 2024-02-05 at 16 22 22@2x](https://github.com/opensumi/core/assets/13938334/907ae60d-e805-4716-bfa4-d3c41a07f001)

**Reply by hanbingzi** [reply]
好的，解决了，确实是clientToken没有配置，谢谢
